packages/donation-form/src/donation-form.ts

```diff
@@ -228,31 +225,44 @@ export class DonationForm extends LitElement {
       return;
     }
 
+    const contactInfo = this.contactForm.donorContactInfo;
+
+    switch (this.selectedPaymentProvider) {
+      case PaymentProvider.CreditCard:
+        this.handleCreditCardDonationFlow(contactInfo);
+        break;
+      case PaymentProvider.Venmo:
+        this.handleVenmoDonationFlow(contactInfo);
+        break;
+    }
+  }
+
+  private async handleCreditCardDonationFlow(contactInfo: DonorContactInfo): Promise<void> {
     const creditCardHandler = await this.braintreeManager?.paymentProviders.creditCardHandler.get();
     creditCardHandler?.hideErrorMessage();
+    const valid = this.contactForm?.reportValidity();
+    const hostedFieldsResponse = await this.paymentFlowHandlers?.creditCardHandler?.tokenizeFields();
 
-    const valid = this.contactForm.validate();
-
-    if (!valid) {
+    if (!valid || hostedFieldsResponse === undefined) {
       return;
     }
 
-    const contactInfo = this.contactForm.donorContactInfo;
     this.emitPaymentFlowStartedEvent();
+    this.donationInfo &&
+      this.paymentFlowHandlers?.creditCardHandler?.paymentInitiated(
+        hostedFieldsResponse,
+        this.donationInfo,
+        contactInfo,
+      );
+  }
 
-    switch (this.selectedPaymentProvider) {
-      case PaymentProvider.CreditCard:
-        this.donationInfo &&
-          this.paymentFlowHandlers?.creditCardHandler?.paymentInitiated(
-            this.donationInfo,
-            contactInfo,
-          );
-        break;
-      case PaymentProvider.Venmo:
-        this.donationInfo &&
-          this.paymentFlowHandlers?.venmoHandler?.paymentInitiated(contactInfo, this.donationInfo);
-        break;
+  private async handleVenmoDonationFlow(contactInfo: DonorContactInfo): Promise<void> {
+    const valid = this.contactForm?.reportValidity();
+    if (!valid) {
+      return;
     }
+    this.donationInfo &&
```

Comment:
flow q: can a user ever be on Venmo flow without donationInfo?

q: I see that `this.donationInfo` is "required" for both payment flows, what happens if they do not meet this condition? or, will they ever not meet this condition?

Reply:
Good catch! I moved this up a level since it's needed in both flows. We should never not have `donationInfo`, but now if we don't, we show an alert.